api/src/DataProvider/TopBookCollectionDataProvider.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\DataProvider;
+
+use ApiPlatform\Core\DataProvider\ArrayPaginator;
+use ApiPlatform\Core\DataProvider\ContextAwareCollectionDataProviderInterface;
+use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
+use App\Entity\TopBook;
+
+final class TopBookCollectionDataProvider implements ContextAwareCollectionDataProviderInterface, RestrictedDataProviderInterface
+{
+    private TopBookDataProvider $dataProvider;
+    private array $collection;
+    private array $context;
+
+    public function __construct(TopBookDataProvider $dataProvider)
+    {
+        $this->dataProvider = $dataProvider;
+    }
+
+    public function supports(string $resourceClass, string $operationName = null, array $context = []): bool
+    {
+        return TopBook::class === $resourceClass;
+    }
+
+    /**
+     * @throws \RuntimeException
+     */
+    public function getCollection(string $resourceClass, string $operationName = null, array $context = []): iterable
+    {
+        $this->context = $context;
+        try {
+            $this->collection = $this->dataProvider->getTopBooks();
+        } catch (\Exception $e) {
+            throw new \RuntimeException(sprintf('Unable to retrieve top books from external source: %s', $e->getMessage()));
+        }
+
+        return new ArrayPaginator($this->collection, $this->getOffset(), $this->getItemsPerPage());
```

Comment:
A better way to handle a pagination would be (for example) to use a custom extension:
1- create a [TopBookCollectionExtensionInterface](https://github.com/vincentchalamon/nav-bundle/blob/master/src/Bridge/ApiPlatform/DataProvider/CollectionExtensionInterface.php)
2- create a [TopBookPaginationExtension](https://github.com/vincentchalamon/nav-bundle/blob/master/src/Bridge/ApiPlatform/DataProvider/Extension/PaginationExtension.php)
3- adapt your [TopBoolCollectionDataProvider](https://github.com/vincentchalamon/nav-bundle/blob/master/src/Bridge/ApiPlatform/DataProvider/CollectionDataProvider.php) to handle the extensions

This extension will help to bridge with the API Platform configuration options, to let the user chose the items-per-page, etc.

Bonus: with this extensions feature, you'll be able to add filters on it, with a TopBookFiltersExtension for example :wink: